### PR TITLE
Improved event scraping

### PR DESCRIPTION
**Pull request by mrkrause**
Hi,

As I mentioned last week on Twitter, I'm excited about this and would like to set up a version at McGill. So...I started doing some coding to make that happen.

- Commits 13922a2 and 58432e5 move the existing scrapers into a Penn-specific module. It adds a config system to pick the scraping module (and a few other configuration things). I was shooting for something that would make it easy to use the same code in multiple places. 
- Commit 59d26db just fixes a minor bug (two functions fetching the same data)
- Commit e12a57f standardizes the event dictionaries so they all have the same fields in the same order. (I was just going over the code, but figured it'd be helpful). 

I hope this helps! Have a good weekend,
--Matt

### CODE REVIEW

**Comment by titipata** [comment]
Overall, this looks great @mrkrause! I will check the PR today and merge if everything works fine!

Titipat A.

**Comment by mrkrause** [comment]
Awesome! Please check the url/pdf stuff carefully, as I don't have GROBID installed yet. I don't think I changed it, but I was moving a bunch of stuff around.

File: backend/fetch_events.py [hunk 1]
```diff
@@ -179,3229 +172,119 @@ def parse_pdf_abstract(pdf_url):
...
-    return events
-
+    df = df.sort_values('event_index', na_position='last')
+    event_index = df.event_index.iloc[0]
+    r = df.iloc[-1].to_dict()
+    r['event_index'] = event_index
+    return r
 
-def fetch_events_cceb(base_url='https://www.cceb.med.upenn.edu/events'):
+def find_scrapers(module) -> List[Callable]:
```
Comment (bluenex) [inline backend/fetch_events.py]: This is very helpful.

@titipata please make sure to review this function.

**Comment by titipata** [comment]
@mrkrause I got a few errors here

- when I try to run `fetch_events.py`, I got this following error:

```
  File "fetch_events.py", line 18, in <module>
    import backend.config as config
ModuleNotFoundError: No module named 'backend'
```

- We just have to fix this [line](https://github.com/mrkrause/penn-events-calendar/blob/master/backend/schedule_fetch_events.py#L24) when running the scheduler.

I will keep you posted with the questions!

**Comment by titipata** [comment]
@mrkrause No worry! So we basically use `supervisor` to manage all the tasks. For the backend part, we just set the location to run directly from the `backend` folder. You can see it [here](https://github.com/mrkrause/penn-events-calendar/blob/master/devops/penn-calendar.supervisor.conf#L212-L220).

File: backend/fetch_events.py (hunk 1 above)
Comment (titipata) [reply backend/fetch_events.py]: I checked and this works perfectly!

**Comment by titipata** [comment]
@mrkrause I'll check the import beyond top-level package quickly. Seems like we might have this specific issue. One way to solve is to use `sys` to add a path from the directory above `local/upenn`.

**Comment by mrkrause** [comment]
Commit 5ccf647f345c68afa57e2302015dc4b722c44161 fixes the module stuff.

**Comment by titipata** [comment]
@mrkrause I think it looks great! Thanks so much again for the PR! I added a minor fix to the code in 12b291e (mainly some bugs that I newly found).

**Comment by titipata** [comment]
@mrkrause This should be in a separate thread. However, I couldn't find your email address. Let us know if you need help with the set up for McGill University. I can provide and information on a machine we use to run the server and some technical details.

cc. @bluenex

**Pull request merged**